prow/pluginhelp/pluginhelp.go

```diff
@@ -0,0 +1,56 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package pluginhelp defines structures that represent plugin help information.
+// These structs are used by sub-packages 'hook' and 'externalplugins'.
+package pluginhelp
+
+// PluginHelp is a serializable representation of the help information for a single plugin.
+// This includes repo specific configuration for every repo that the plugin is enabled for.
+type PluginHelp struct {
+	// Description is a description of what the plugin does and what purpose it achieves.
+	Description string
+	// WhoCanUse is a description of the permissions/role/authorization required to use the plugin.
+	// This is usually specified as a github permissions, but it can also be a github team, an
+	// OWNERS file alias, etc.
```

Comment:
It seems to me that we need to document to the full extend what's supported here. Not sure what else is supported by reading `etc.`. Also what happens if a plugin can be used by more than one github team? How do I specify that here?

Reply:
Any feature a plugin can use to differentiate users will work. There isn't a limited set of permission types that we support. Currently only github repo permissions and github teams are used to identify users who can use the plugins, but there is nothing to stop plugins from using OWNERS file aliases, slack membership, or any other differentiating feature.
The field is just a string so I expected plugins to create a one sentence description of who can use the plugin. If multiple github teams can use a plugin, this field will say exactly that. If the specific teams differ by repo, then that can be elaborated in the per repo config for the plugin: `PluginHelp.Config`.

Reply:
Yes, this is fine. I thought this would be used programmatically somehow.